src/components/operations/operation-details/ko/runtime/authorization.ts

```diff
@@ -146,6 +146,7 @@ export class Authorization {
         keyHeader.inputTypeValue("password");
         keyHeader.type = "string";
         keyHeader.required = true;
+        keyHeader.isReadOnly = true;
```

Comment:
If they want to specify a key, they can add a new header with name 'Authorization' and specify the key there. Adding the header like this won't make it read-only, since the header is not added through this method.
This method is only called by the sign flow, so the value for the Authorization header, in this case, is generated by a successful sign in through OAuth. 
Is it a valid case when users sign-in through OAuth, the header is added and then they need to specify a different key?

Reply:
How about the case when `Authorization` header present in API definition and marked as required? Seems like it will be rendered, but users won't be able to modify its value. Same goes to `Ocp-Apim-Subscription-Key`. Thoughts?

Reply:
If the Authorization header comes from the API definition and is marked as required, it won't be added through this method: it is mapped as a consoleHeader in consoleRequest's constructor, so it is not marked as read-only (it's treated as any other header).

Regarding the Ocp-Apim-Subscription-Key header, it's the same behavior: its value is "generated" based on what is introduced in the Subscription Key textbox in the Authorization section (or selected from the combo box). If the user is not signed-in (and have no subscriptions), they can still specify the key there, and the header value gets updated accordingly. 
If the header was set in the API definition, it will be treated as a normal header (added as not read-only).

Since they are basically generated values, should we allow editing them?

Reply:
However, I did miss another case with this logic:
If the user signs in through OAuth, the Authorization header is set and it is read-only. If then, they go to the authorization section and select "No auth" from the combo box, the header value will be cleared, but the header still remains displayed and it is read-only, which is indeed confusing.
![no-auth](https://user-images.githubusercontent.com/92857141/183386680-2c98251e-726b-46fa-9524-a4d67173f6b3.gif)

 So we can either:
1. Remove the read-only authorization header when the grant type is set to undefined (similar to how we remove  Ocp-Apim-Subscription-Key when there is no key specified) or
2. Leave it editable as it is now. (or we can leave both of them editable as it is now and remove the "read-only" logic)

What do you think?

Reply:
But I think you are right, maybe after successful sign-in with OAuth, they want to test it with a different key generated by some different authorization and, in that case, it would be easier to just edit the already existing header instead of adding a new one.

Removed it